### PR TITLE
Fixed a crash when using logger in an environment with no plugins loaded

### PR DESCRIPTION
**Pull request by przemyslaw-dobrowolski-cl**

### CODE REVIEW

File: extension/goplugin/environment_test.go [hunk 1]
```diff
@@ -50,6 +50,16 @@ var _ = Describe("Environment", func() {
 		env = goplugin.NewEnvironment("test", nil, nil)
 	})
 
+	Describe("Internal state consistency", func() {
```
Comment (p-kozlowski) [inline extension/goplugin/environment_test.go]: still failes if Schemas() returns empty list:
```
13:43:22.906 extest CRITICAL  failed to load schemas extensions: failed to init db: No schemas in Manager. Did you remember to load them?
```

File: extension/goplugin/transaction_test.go [hunk 2]
```diff
@@ -68,7 +68,12 @@ var _ = Describe("Transaction", func() {
 			testSchema = env.Schemas().Find("test")
 			Expect(testSchema).To(Not(BeNil()))
 
-			err = db.InitDBWithSchemas(DbType, DbFile, true, true, false)
+			err = db.InitDBWithSchemas(DbType, DbFile, db.InitDBParams{
```
Comment (p-kozlowski) [inline extension/goplugin/transaction_test.go]: there's some suspicious inconsistency here: in some tests it's true/false/false/true, in others false/false/false/true and here it's true/true/false/true. why is that?
could we add some universal safe defaults? eg.
```
func DefaultTestDBParams() InitDbParams
```

Comment (przemyslaw-dobrowolski-cl) [reply extension/goplugin/transaction_test.go]: ok, fixed

**Comment by przemyslaw-dobrowolski-cl** [comment]
rebased

File: extension/goplugin/transaction_test.go [hunk 3]
```diff
@@ -67,9 +67,7 @@ var _ = Describe("Transaction", func() {
 			Expect(env.Start()).To(Succeed())
 			testSchema = env.Schemas().Find("test")
 			Expect(testSchema).To(Not(BeNil()))
-
-			err = db.InitDBWithSchemas(DbType, DbFile, true, true, false)
-			Expect(err).To(BeNil())
+			Expect(db.InitDBWithSchemas(DbType, DbFile, db.DefaultTestInitDBParams())).To(BeNil())
```
Comment (p-kozlowski) [inline extension/goplugin/transaction_test.go]: minor: Expect(...).To(Succeed())

Comment (przemyslaw-dobrowolski-cl) [reply extension/goplugin/transaction_test.go]: fixed

**Comment by przemyslaw-dobrowolski-cl** [comment]
rebased

File: extension/goplugin/environment_test.go (hunk 1 above)
Comment (przemyslaw-dobrowolski-cl) [reply extension/goplugin/environment_test.go]: it should be already fixed

**Pull request merged**